docs/src/kernel/linux-compatibility/limitations-on-system-calls/memory-management.md

```diff
@@ -65,3 +70,131 @@ Unsupported flags:
 
 For more information,
 see [the man page](https://man7.org/linux/man-pages/man2/mmap.2.html).
+
+### `msync`
+
+Supported functionality in SCML:
+
+```c
+// Flush memory region to disk asynchronously with cache invalidation
+msync(
+    addr, length,
+    flags = MS_ASYNC | MS_INVALIDATE
+);
+
+// Flush memory region to disk synchronously with cache invalidation
+msync(
+    addr, length,
+    flags = MS_SYNC | MS_INVALIDATE
+);
+```
+
+Silently-ignored flags:
+* `MS_INVALIDATE` is ignored because all processes use the same page cache
+
+For more information,
+see [the man page](https://man7.org/linux/man-pages/man2/msync.2.html).
+
+### `mremap`
+
+Supported functionality in SCML:
+
+```c
+// Remap a memory region allowing kernel relocation
+mremap(
+    old_address,
+    old_size,
+    new_size,
+    flags = MREMAP_MAYMOVE
+);
+
+// Remap a memory region to specific address with forced relocation
+mremap(
+    old_address,
+    old_size,
+    new_size,
+    flags = MREMAP_MAYMOVE | MREMAP_FIXED,
+    new_address
+);
+```
+
+For more information,
+see [the man page](https://man7.org/linux/man-pages/man2/mremap.2.html).
+
+## `Virtual Memory Operations`
+
+### `mprotect`
+
+Supported functionality in SCML:
+
+```c
+// Set memory access permissions
+mprotect(
+    addr,
+    len,
+    prot =
+        PROT_NONE |  // Disable all access
+        PROT_READ |  // Enable read  (r--)
+        PROT_WRITE | // Enable write (-w-)
+        PROT_EXEC    // Enable execute (--x)
+);
+```
+
+Silently-ignored protection flags:
+* `PROT_SEM`
+* `PROT_SAO`
+* `PROT_GROWSUP`
+* `PROT_GROWSDOWN`
+
+For more information,
+see [the man page](https://man7.org/linux/man-pages/man2/mprotect.2.html).
+
+### `madvise`
+
+Supported functionality in SCML:
+
+```c
+// Use default memory access pattern (no special optimization)
+madvise(
+    addr,
+    length,
+    advice = MADV_NORMAL
+);
+
+// Sequential access expected: aggressive read-ahead and immediate page release
+madvise(
+    addr,
+    length,
+    advice = MADV_SEQUENTIAL
+);
+
+// Prefetch pages for near-future access
```

Comment:
Let's follow Rust's convention to start with a verb in the documentation of a function.